src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1247,14 +1224,52 @@ void IThreadPoolWorkItem.MarkAborted(ThreadAbortException tae)
             MarkExecuted(true);
 #endif
         }
+    }
+
+    internal class QueueUserWorkItemCallbackWithState : QueueUserWorkItemCallback
```

Comment:
I don't think the callback+state version is particularly valuable.  In pretty much any case I can think of, someone who cares about performance is going to be passing state, since it's how you use a cached delegate repeatedly.  I would simply have a version with callback+state and then another that adds context.